packages/api/src/routes/middlewares/connect-device.ts

```diff
@@ -0,0 +1,73 @@
+import { getConnectedUserOrFail } from "../../command/connected-user/get-connected-user";
+import { updateProviderData } from "../../command/connected-user/save-connected-user";
+import { getUserToken } from "../../command/cx-user/get-user-token";
+import UnauthorizedError from "../../errors/unauthorized";
+import { ConnectedUser } from "../../models/connected-user";
+import { RPMDeviceProviderOptions } from "../../shared/constants";
+
+/**
+ * Connects the user to a provider that does not use OAuth1 or OAuth2.
+ *
+ * @param {RPMDeviceProviderOptions}      provider      A medical device provider (i.e. Tenovi)
+ * @param {string}                        token         Connect Token
+ * @param {string}                        deviceIds     Comma-separated string of device IDs
+ * @param {string}                        deviceUserId  Device User ID
+ * @returns
+ */
+export const connectDevice = async (
+  provider: RPMDeviceProviderOptions,
+  token: string,
+  deviceIds: string,
+  deviceUserId: string
+): Promise<ConnectedUser> => {
+  const userToken = await getUserToken({ token });
+  const cxId = userToken.cxId;
+  const userId = userToken.userId;
+
+  if (!provider || !cxId || !userId) {
+    throw new UnauthorizedError();
+  }
+
+  const connectedUser = await getConnectedUserOrFail({ id: userId, cxId });
+
+  const connectedDeviceIds = getConnectedDeviceIds(connectedUser, provider, deviceIds);
+
+  // save the access token in the provider map
+  const updatedConnectedUser = await updateProviderData({
+    id: userId,
+    cxId,
+    provider,
+    providerItem: {
+      token: "true",
+      connectedDeviceIds,
+      deviceUserId,
+    },
+  });
+
+  return updatedConnectedUser;
+};
+
+/**
+ * Gets the IDs of all connected devices for the user.
+ *
+ * @param connectedUser  Connected User
+ * @param provider       A healthcare provider
+ * @param deviceIds      A comma-separated string of devices IDs
+ * @returns
+ */
+function getConnectedDeviceIds(
```

Comment:
Good catch!

Reply:
I think the name of the function makes sense. I'll change the description to reflect the merging of connected devices into a list. But since it returns a list, I think `get...` makes more sense for the name